lib/util.js

```diff
@@ -504,13 +504,11 @@ function getPrefix(constructor, tag, fallback) {
     return `${constructor} `;
   }
 
-  if (tag !== '')
-    return `[${tag}] `;
-
-  if (fallback !== undefined)
-    return `${fallback} `;
+  if (tag !== '' && constructor !== tag) {
+    return `[${fallback}: null prototype] [${tag}] `;
+  }
```

Comment:
👆 These are just inferences for a `null` prototype and could be falsy flagged as such. 
Could we not just check `Reflect.getPrototypeOf(value) === null`?

Reply:
@jdalton do you have an example that would fail the current check?

Reply:
@BridgeAR 

It looks like by way of the check that an object with a falsey `Symbol.toStringTag` value and an empty `.name` property on the constructor will be considered as having a `null` prototype.

Reply:
After looking at it again, I agree that it's best to add another check. Here is a test case that should not be detected as null prototype:

```js
const obj = [1, 2]
const value = (function() {return function() {}})()
Object.setPrototypeOf(obj, value)
```

I suggest to do the following:

In the `getConstructorName` function add `if (obj === null) { return null; }` after the while loop (https://github.com/nodejs/node/pull/22331/files#diff-3deb3f32958bb937ae05c6f3e4abbdf5L495).

The constructor can be `null` in that case and we have to check that in the `getPrefix` function from then on as well.
That should likely be the fasted way to achieve that with little churn.

Reply:
@BridgeAR: Sorry for being late here. 

The test case that you had posted:

```
const obj = [1, 2]
const value = (function() {return function() {}})()
Object.setPrototypeOf(obj, value)
```

actually prints the output as expected:

```
{ '0': 1, '1': 2 }
```

Also, I guess we are calling `getPrefix` only after deciding whether the prototype is null or not. For example, in `noPrototypeIterator` we first check `Object.getPrototypeOf(value)` and then wrap with null sub class. Am I missing something here?

Reply:
Well, yes `getPrefix` can be tweaked to give falsy checks and return null prototype string. Let me see if I can pull out an example for the same.

Reply:
Could you add a test that mimics what would trigger a false check to help catch changes in flow?

Reply:
Hmmm, sorry I couldn't able to pull out a test case that would trigger a false check. 

Need your help here. @BridgeAR Any thoughts / ideas please?

Reply:
@BridgeAR Ping. I couldn't able to pull out a test case, which will provide falsy check from `getPrefix` (may be we are good enough as described above in my comments). Will you be able to help me to find a case if one possible? Thanks..

Reply:
My example did not work because it's an array and when changing the prototype of the array to a function, it won't contain the `Symbol.iterator` anymore and we do not yet have a special handling manipulated symbols / cases like these. I also missed to set the prototype of the prototype to null.

So changing it to an object and adding the null prototype shows the wrong output as expected: 

```js
const obj = { a: true }
const value = (function() {return function() {}})()
Object.setPrototypeOf(value, null)
Object.setPrototypeOf(obj, value)
console.log(obj)
// [Object: null prototype] { a: true }
assert.notStrictEqual(Object.getPrototypeOf(obj), null)
// No assertion
```

Reply:
@BridgeAR Oh alright. I didn't thought through setting prototype of the prototype to null. Thats a good catch. Will work on the fix and update the PR.

Reply:
I could just push another commit that addresses everything, if you do not mind.

Reply:
This should now be addressed with my commit.